openslides_backend/action/actions/user/user_mixin.py

```diff
@@ -238,3 +238,155 @@ def check_meeting_and_users(
+        all_groups_added: Set[int] = set()
+        all_groups_removed: Set[int] = set()
+        info_cml_or_oml = False
+        info_active: Set[bool] = set()
+
+        # Scan the instances and collect the info for the history information
+        for instance in self.instances:
+            instance_fields = set(instance.keys())
+
+            update_fields = set(
+                [
+                    "title",
+                    "first_name",
+                    "last_name",
+                    "email",
+                    "username",
+                    "default_structure_level",
+                    "default_number",
+                    "default_vote_weight",
+                ]
+            )
+            if instance_fields & update_fields:
+                info_update = True
+
+            meeting_ids: Set[str] = set()
+            for field in ("structure_level_$", "number_$", "vote_weight_$"):
+                if field in instance:
+                    meeting_ids.update(instance[field] or set())
+            all_meetings.update(meeting_ids)
+
+            if "group_$_ids" in instance:
+                group_ids_from_instance = self.get_group_ids_from_instance(instance)
+                group_ids_from_db = self.get_group_ids_from_db(instance)
+                added = group_ids_from_instance - group_ids_from_db
+                removed = group_ids_from_db - group_ids_from_instance
+                all_groups_added.update(added)
+                all_groups_removed.update(removed)
+
+            if (
+                "organization_management_level" in instance
+                or "committee_$_management_level" in instance
+            ):
+                info_cml_or_oml = True
+            if "is_active" in instance:
+                info_active.add(instance["is_active"])
+
+        # add the different history information
+        if info_update:
+            informations.append("User updated")
+
+        if any(all_meetings):
+            self.add_to_history(
+                informations,
+                all_meetings,
+                "User updated in meeting {}",
+                "User updated in multiple meetings",
+                "meeting",
+            )
+
+        check_added = any(all_groups_added)
+        check_removed = any(all_groups_removed)
+        if check_added and check_removed:
+            informations.append("Groups changed")
+        elif check_added:
+            self.add_to_history(
+                informations,
+                all_groups_added,
+                "Group {} added",
+                "Groups changed",
+                "group",
+            )
+        elif check_removed:
+            self.add_to_history(
+                informations,
+                all_groups_removed,
+                "Group {} removed",
+                "Groups changed",
+                "group",
+            )
+
+        if info_cml_or_oml:
+            informations.append("OML/CML changed")
```

Comment:
Ah, missed that before, OML and CML should be separate messages

Reply:
Done. Split OML and CML.